shared/reducers/chat.js

```diff
@@ -376,8 +376,12 @@ function reducer (state: State = initialState, action: Actions) {
       const toFind = convo.get('conversationIDKey')
       const oldInbox = state.get('inbox')
       const existing = oldInbox.findEntry(i => i.get('conversationIDKey') === toFind)
-      const updatedInbox = existing ? oldInbox.set(existing[0], convo) : oldInbox.push(convo)
-      return state.set('inbox', sortInbox(updatedInbox))
+      let updatedInbox = existing ? oldInbox.set(existing[0], convo) : oldInbox.push(convo)
+      // time changed so we need to sort
+      if (!existing || existing[1].time !== convo.get('time')) {
+        updatedInbox = sortInbox(updatedInbox)
```

Comment:
only sort if the timestamp of something changed. we were burning a lot of cpu when this isn't the case